src/components/data-exploration/generic-gene-table/SelectionActions.jsx

```diff
@@ -97,6 +98,21 @@ const SelectionActions = (props) => {
             {listed ? 'Hide' : 'List'}
           </Button>
           <ComponentActions name='Heatmap' experimentId={experimentId} componentType={COMPONENT_TYPE} />
+          <Button
+            type='link'
+            size='small'
+            onClick={() => setCreateCellSetModalVisible(!createCellSetModalVisible)}
+          >
+            Cellset
+          </Button>
+          {
+            createCellSetModalVisible && (
+              <CreateCellSetModal
+                selectedGenes={selectedGenes}
+                onCancel={() => setCreateCellSetModalVisible(false)}
+              />
+            )
+          }
```

Comment:
I wouldn't have this code in `SelectionActions` - it is not the job of `SelectionActions` to know when the Cellset button should be visible or not. This logic should be handled inside `CreateCellSetModal`. Also, this `setCreateCellSetModalVisible` variable is a duplicate of `selectedGenes.length !== 0`, which is checked at the beginning of the return statement in `SelectionActions`.
You can delete this `createCellSetModalVisible` state variable and follow the concept behind the implementation of of `ComponentActions`